src/session_info2/__init__.py

```diff
@@ -41,9 +44,49 @@ def _cpu_info() -> str:
         proc = platform.processor() or None
         return f"{cpu_count()} logical CPU cores{f', {proc}' if proc else ''}"
 
+    @staticmethod
+    def _gpu_info() -> str | list[str]:
+        """Get GPU info."""
+        if platform.system() == "Windows":
+            # If the platform is Windows and nvidia-smi
+            # could not be found from the environment path,
+            # try to find it from system drive with default installation path
+            nvidia_smi = shutil.which("nvidia-smi")
+            if nvidia_smi is None:
+                postfix = "\\Program Files\\NVIDIA Corporation\\NVSMI\\nvidia-smi.exe"
+                nvidia_smi = f"{os.environ['SYSTEMDRIVE']}{postfix}"
+        else:
+            nvidia_smi = "nvidia-smi"
+
+        # Get ID, processing and memory utilization for all GPUs
+        try:
+            p = Popen(  # noqa: S603
+                [
+                    nvidia_smi,
+                    "--query-gpu=index,name,memory.total,driver_version",
+                    "--format=csv,noheader,nounits",
+                ],
+                stdout=PIPE,
+            )
+            stdout, _ = p.communicate()
+        except:  # noqa: E722
+            return "No GPU found"
+        output = stdout.decode("UTF-8")
+
+        # Split on line break
+        lines = output.split(os.linesep)
+        numdevices = len(lines) - 1
+        gpus = list()
+        for g in range(numdevices):
+            line = lines[g]
+            vals = line.split(", ")
+            gpus.append(f"ID: {vals[0]}, {vals[1]}, Driver:{vals[3]}, Memory:{vals[2]}")
+        return gpus
```

Comment:
1. Never do `for index in range(numdevices)` and then use the index to access elements in a list. Just loop over an iterable instead.
2. prefer comprehensions to `append`
3. use `name1, name2, name3, name4 = line.split(...)` instead of `vals = line.split(...)`. (and of course give the variables good names)

like:

```py
device_infos = (line.split(", ") for line in lines[:numdevices])
return [
    f"ID: {id}, {whatever_this_is}, Driver: {driver}, Memory: {memory}"
    for id_, whatever_this_is, memory, driver in device_infos
]
```